libs/libcommon/src/libcommon/croissant_utils.py

```diff
@@ -126,16 +126,24 @@ def feature_to_croissant_field(
             ],
         }
     elif isinstance(feature, (LargeList, list, Sequence)):
+        array_shape = []
         if isinstance(feature, list):
             if len(feature) != 1:
                 return None
             sub_feature = feature[0]
+            array_shape.append("-1")
         else:
+            if shape := feature.length:
```

Comment:
nit: For readability, collect `int | None` instead of str, and clean afterwards when building the actual arrayShape.

```python
array_shape.append(-1)
...
array_shape.append(feature.length)
...
array_shape.append(sub_feature.length)
...

str_array_shape = [str(shape) if shape else '-1' for shape in array_shape]
field["arrayShape"] = ",".join(str_array_shape)
```

Reply:
Good point, done.